alternator/executor.cc

```diff
@@ -96,6 +96,44 @@ std::string make_jsonable::to_json() const {
     return rjson::print(_value);
 }
 
+
+static bool is_leaf(const rjson::value& value) {
+    return !value.IsObject() && !value.IsArray();
+}
+
+static future<> destroy_gently_nonleaf(rjson::value&& value_in);
+
+future<> destroy_gently(rjson::value&& value) {
+    if (is_leaf(value)) {
+        return make_ready_future<>();
+    } else {
+        return destroy_gently_nonleaf(std::move(value));
+    }
+}
+
+static
+future<>
+destroy_gently_nonleaf(rjson::value&& value_in) {
+    // value_in is an rvalue ref to force caller to move
+    auto value = std::move(value_in);
```

Comment:
What does this line achieve? The comment seems to explain why value_in was defined the way it was, not why you wanted to move it again here.

Reply:
We move value_in to the coroutine frame. I'll expand the comment, though it will look like "add 1 to i".